File: consumer.js
const { Kafka } = require('kafkajs');

const kafka = new Kafka({
  clientId: 'my-consumer',
  brokers: ['localhost:9092']
});

const consumer = kafka.consumer({ groupId: 'test-group' });

const run = async () => {
  await consumer.connect();
  await consumer.subscribe({ topic: 'test-topic', fromBeginning: true });

  await consumer.run({
    eachMessage: async ({ message }) => {
      console.log(`Received: ${message.value.toString()}`);
    }
  });
};

run().catch(console.error);

File: consumer.py
# consumer.py
from kafka import KafkaConsumer

def main():
    consumer = KafkaConsumer(
        'test-topic',
        bootstrap_servers=['localhost:9092'],
        auto_offset_reset='earliest',
        enable_auto_commit=True,
        group_id="my-group",
        value_deserializer=lambda x: x.decode('utf-8')  # just decode string, no JSON
    )

    print("🚀 Consumer is running... Waiting for messages.")
    
    try:
        for message in consumer:
            print(f"Received: {message.value}")
    except KeyboardInterrupt:
        print("\n❌ Stopped by user")
    finally:
        consumer.close()

if __name__ == "__main__":
    main()
